lib/langs/python.rb

```diff
@@ -67,10 +94,56 @@ def parse_pipfile(content)
     def latest_version(name)
       content = HTTP.follow(max_hops: 1).get(::File.join(API_URL, name, 'json'))
       begin
-        Gem::Version.new(JSON.parse(content)['info']['version'])
-      rescue
+        v = JSON.parse(content)['info']['version']
+        Gem::Version.new(canonical_version(v))
+      rescue StandardError => e
+        log_error("Encountered error when fetching latest version of #{name}: #{e.class} #{e.message}")
         nil
       end
     end
+
+    private
+
+    def get_type(type)
+      type == 'dev-packages' ? :development : :runtime
+    end
+
+    def canonical_version(v)
+      case v
```

Comment:
I don't know if the case statement makes much sense here. Ternary version would work just as well:
```ruby
version.match(/^\./) ? "0#{version}" : version
```